src/mindlessgen/qm/orca.py

```diff
@@ -170,13 +199,102 @@ def _run(self, temp_path: Path, arguments: list[str]) -> tuple[str, str, int]:
             orca_log_err = e.stderr.decode("utf8", errors="replace")
             return orca_log_out, orca_log_err, e.returncode
 
+    def _run_xtb_driver(
+        self,
+        temp_path: Path,
+        geometry_filename: str,
+        xcontrol_name: str,
+        ncores: int,
+    ) -> tuple[str, str, int]:
+        """
+        Run the optimization through the xTB external driver when constraints are requested.
+        """
+        xtb_executable = self._get_xtb_executable()
+        arguments = [
+            str(xtb_executable),
+            geometry_filename,
+            "--opt",
+        ]
+        opt_level = getattr(self.cfg, "optlevel", None)
+        if opt_level not in (None, ""):
+            arguments.append(str(opt_level))
+        arguments.extend(["--orca", "-I", xcontrol_name])
+        try:
+            xtb_out = sp.run(
+                arguments,
+                cwd=temp_path,
+                capture_output=True,
+                check=True,
+            )
+            xtb_log_out = xtb_out.stdout.decode("utf8", errors="replace")
+            xtb_log_err = xtb_out.stderr.decode("utf8", errors="replace")
+            return xtb_log_out, xtb_log_err, 0
+        except sp.CalledProcessError as e:
+            xtb_log_out = e.stdout.decode("utf8", errors="replace")
+            xtb_log_err = e.stderr.decode("utf8", errors="replace")
+            return xtb_log_out, xtb_log_err, e.returncode
+
+    def _get_xtb_executable(self) -> Path:
+        """
+        Determine the path to the xTB executable for external ORCA optimizations.
+        """
+        for attr_name in ("xtb_driver_path", "xtb_path"):
```

Comment:
to be more DRY:
```suggestion
        candidates = (getattr(self.cfg, attr_name, None) for attr_name in ("xtb_driver_path", "xtb_path") if getattr(self.cfg, attr_name, None) is not None)
        candidate = next(candidates, None)
        try:
            return get_xtb_path(candidate)
        except ImportError as exc:
            ...
```

btw i was wondering why there are two attributes to store the xtb path? imo it should only be one (`xtb_path`)

Reply:
+1

Reply:
Right it is now only `xtb_path`